### ISSUE TITLE
Match ES2022 Error `cause` property on AxiosError

### ISSUE THREAD

**Issue by ericdowell** [axios/axios#5849]
### Describe the bug

When trying to assign `AxiosError.cause` to `unknown` I get the following error, which does not happen with a normal `Error` class.

```
TS2739: Type  {}  is missing the following properties from type  Error :  name, message 
```

### To Reproduce

Passing in `ErrorOptions` to a helper function we can recreate the issue. Currently I will need to add `// @ts-ignore` to get around this issue.

### Code snippet

```js
export const setCauseError = (
  error: AxiosError,
  options?: ErrorOptions
): Error => {
  if (options?.cause) {
    error.cause = options.cause
  }
  return error
}
```

### Expected behavior

No TS error should be emitted to my IDE/type checking.

### Axios Version

1.5.0

### Adapter Version

n/a

### Browser

n/a

### Browser Version

n/a

### Node.js Version

18.16.0

### OS

macOS 13.5.1

### Additional Library Versions

```bash
typescript 5.1.6
```

### Additional context/Screenshots

With `Error`:

<img width="720" alt="Screenshot 2023-08-31 at 1 37 08 PM" src="https://github.com/axios/axios/assets/525506/3c3a32a9-432c-4274-bd7a-0a8e0e5142d0">

With `unknown`:

<img width="401" alt="Screenshot 2023-08-31 at 1 37 39 PM" src="https://github.com/axios/axios/assets/525506/3e829416-5b89-4c58-91a9-0eb49ed2583d">

**Comment by ericdowell** [comment]
@jasonsaayman Are you or another member that maintains this package to take a look at the PR I've opened? https://github.com/axios/axios/pull/5850

**Comment by jasonsaayman** [comment]
yes i will as soon as possible but im currently just tryint to reign in the issues and then move onto PR's

**Comment by ericdowell** [comment]
Change was applied in commit https://github.com/axios/axios/commit/78b290c57c978ed2ab420b90d97350231c9e5d74 on Jul 30, 2025. [PR auto closed](https://github.com/axios/axios/pull/5850#event-20869570271) once [rebased](https://github.com/axios/axios/pull/5850#event-20869570399)

Thanks for checking in @jasonsaayman ❤️!
